Fix team page - populate with current members

bryanzhang3 opened the pull request:
## Summary
- Imported 32 current members from Notion to Contentful
- Organized team page by team sections with headers
- Fixed director display logic

## Changes
- Added Contentful migration scripts
- Reorganized team page UI to group members by teams
- Added default avatar for all members

Fixes #183

bryanzhang3 committed:
Fix team page - populate with current members

- Imported 32 current members from Notion to Contentful
- Organized team page by team sections with headers
- Fixed director display logic to show directors in their teams
- Added Contentful migration scripts for future updates
- Added default avatar for all members

Fixes #183

tr-vs commented on src/app/team/TeamContent/TeamContent.tsx:
```diff
@@ -122,54 +146,34 @@ export default function TeamContent({ members, alumni }: { members: Array<any>,
-        <div className="p-12 grid gap-12 md:grid-cols-2 lg:grid-cols-3 xl:grid-cols-4 md:gap-8 lg:gap-12 xl:gap-16">
-          { alumniList?.filter((data) => data.enabled).map((data, idx) => 
-            {
-              const alumniData = data.fields;
```
It seems like the alumni section is wiped? I don't see it in the preview either--how hard is it to add the section back using the same contentful logic?

bryanzhang3 replied:
Yeah that's my fault I removed it because we don't have reliable alumni data yet but Kate and Angelina are working on it. I added it back so because it's probably more convenient to have it there and just update when we have proper alumni data. It will show incomplete alumni data currently though.

tr-vs commented on scripts/cleanup-test.js:
Not sure if we need to commit these scripts since migration should be a one time thing--feel free to correct me if I'm wrong

bryanzhang3 replied:
I think it may need to be run each year with the csv of the new members on notion. I also just updated publish-members.js to publish only people in the CSV. But if it's too much clutter we could store it somewhere else. I'm not too sure what do you think?

tr-vs replied:
Contentful should be a temporary solution (as we implement the membership portal, it will pull from Supabase instead), so these scripts don't need to be ran yearly

bryanzhang3 replied:
I forgot about that, I removed the scripts

tr-vs commented on src/app/team/TeamContent/TeamContent.tsx:
```diff
@@ -1,12 +1,12 @@
 function MemberCard({ memberData, alumni, className } : { memberData: any, alumni: boolean, className?: string }) {
   const { name, year, titles, degree, fact, company, website} = memberData;
-  const photoURL = memberData?.photo?.fields?.file?.url;
+  // Use default avatar for all members (professional headshots coming soon)
```
Is the reason why we aren't using the images in contentful because some people have a picture set and some don't?

bryanzhang3 replied:
Yes, especially new members. Also I don't know if people know what photo they have in contentful so we should probably ask for them to send us a photo of what they want on the website. But yeah we can pull image from contentful once that's all set up.

tr-vs replied:
Got it, can we make it so that we use the image in contentful, else use the default?

bryanzhang3 replied:
Done

tr-vs reviewed (commented):
Looks good! Went beyond what I expected! I left some comments, please review them! o7

bryanzhang3 committed:
Add alumni section back and improve publish script

- Restored alumni section with filters (data being updated by team)
- Updated publish-members.js to only publish members from CSV
- Prevents accidentally publishing all draft entries

bryanzhang3 committed:
Use Contentful photo if available, else default avatar

- Remove migration scripts (temporary until Supabase migration)

tr-vs commented:
beautiful!
